ext/RMagick/rmimage.c

```diff
@@ -4571,12 +5059,21 @@ VALUE
 Image_delete_compose_mask(VALUE self)
 {
     Image *image;
+#if defined(IMAGEMAGICK_7)
+    ExceptionInfo *exception;
+#endif
 
     image = rm_check_frozen(self);
 
-    // Store a clone of the mask image
+#if defined(IMAGEMAGICK_7)
+    exception = AcquireExceptionInfo();
+    (void) SetImageMask(image, CompositePixelMask, NULL, exception);
+    CHECK_EXCEPTION()
+    (void) DestroyExceptionInfo(exception);
```

Comment:
You would need a method (with a different name) for each signature so I don't think that would feasible here. And I don't think it would make the code very readable.

p.s. After IM7 has been merged I would really like to get rid of the `CHECK_EXCEPTION` defined because it is not clear what it does when you read the code like this. But that will be a discusion for another moment.

Reply:
I agree with removing `CHECK_EXCEPTION`. I've been reading it like a function call when it is a define, and it doesn't really do a whole lot for us. In general, I'm beginning to feel like `#define` should be avoided when possible, unless it provides a *lot* of value.